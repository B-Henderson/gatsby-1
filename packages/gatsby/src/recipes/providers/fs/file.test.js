const Joi = require(`@hapi/joi`)
const file = require(`./file`)
const resourceSchema = require(`../resource-schema`)

const root = __dirname
const content = `Hello, world!`

describe(`file resource`, () => {
  test(`create writes a file`, async () => {
    const filePath = `bar.txt`

    await file.create({ root }, { path: filePath, content })

    const result = await file.read({ root }, { path: filePath })

    expect(result.content).toEqual(content)

    await file.destroy({ root }, { path: filePath })
  })

  test(`update writes a file`, async () => {
    const filePath = `bar.txt`
    const newContent = `new content!`

    await file.create({ root }, { path: filePath, content })
    await file.update({ root }, { path: filePath, content: newContent })

    const result = await file.read({ root }, { path: filePath })

    expect(result.content).toEqual(newContent)

    await file.destroy({ root }, { path: filePath })
  })

  test(`plan returns a description`, async () => {
    const filePath = `file.txt`
    await file.create({ root }, { path: filePath, content })

    const result = await file.plan({ root }, { path: filePath, content })

    expect(result.describe).toEqual(`Write ${filePath}`)

    await file.destroy({ root }, { path: filePath })
  })

  test(`e2e file resource test`, async () => {
    const filePath = `file.txt`
    const createPlan = await file.plan({ root }, { path: filePath, content })
    expect(createPlan).toMatchSnapshot()
    const createResponse = await file.create(
      { root },
      { path: filePath, content }
    )
    const validateResult = Joi.validate(createResponse, {
      ...file.validate(),
      ...resourceSchema,
    })
    expect(validateResult.error).toBeNull()

    expect(createResponse).toMatchSnapshot()
    const readResponse = await file.read({ root }, { path: filePath })
    expect(readResponse).toMatchSnapshot()
    const updatePlan = await file.plan(
      { root },
      { path: filePath, content: content + `1` }
    )
    expect(updatePlan).toMatchSnapshot()
    const updateResponse = await file.update(
      { root },
      { path: filePath, content: content + `1` }
    )
    expect(updateResponse).toMatchSnapshot()
    const destroyReponse = await file.destroy({ root }, { path: filePath })
    expect(destroyReponse).toMatchSnapshot()
  })
})
