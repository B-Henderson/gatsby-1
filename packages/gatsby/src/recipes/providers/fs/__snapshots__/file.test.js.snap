// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`file resource e2e file resource test 1`] = `
Object {
  "currentState": Object {
    "content": "",
  },
  "describe": "Write file.txt",
  "newState": "Hello, world!",
}
`;

exports[`file resource e2e file resource test 2`] = `
Object {
  "content": "Hello, world!",
  "id": "/Users/kylemathews/programs/gatsby/packages/gatsby/src/recipes/providers/fs/file.txt",
  "path": "/Users/kylemathews/programs/gatsby/packages/gatsby/src/recipes/providers/fs/file.txt",
}
`;

exports[`file resource e2e file resource test 3`] = `
Object {
  "content": "Hello, world!",
}
`;

exports[`file resource e2e file resource test 4`] = `
Object {
  "currentState": Object {
    "content": "Hello, world!",
  },
  "describe": "Write file.txt",
  "newState": "Hello, world!1",
}
`;

exports[`file resource e2e file resource test 5`] = `
Object {
  "content": "Hello, world!1",
  "id": "/Users/kylemathews/programs/gatsby/packages/gatsby/src/recipes/providers/fs/file.txt",
  "path": "/Users/kylemathews/programs/gatsby/packages/gatsby/src/recipes/providers/fs/file.txt",
}
`;

exports[`file resource e2e file resource test 6`] = `undefined`;
